### PR TITLE
Clothing Booth Fixes Yay :D

### PR DESCRIPTION
**Pull request by AwkwardDryad**
[bug-minor]

## About the PR 
- Fixed preview icon rendering
- Fixed animation timing
- Fixed AIs and ghosts being able to use the booth
- Fixed spam teleportation upon exiting the booth
- Fixed infinite range clothing booth click

## Why's this needed?
- Clothing booth machine broke.

## Changelog
```
(u)AwkwardDryad:
(+)Fixed tons of clothing booth bugs :D
```

### CODE REVIEW

File: code/modules/vending/clothingbooth.dm [hunk 1]
```diff
@@ -65,6 +62,9 @@ var/list/clothingbooth_items = list()
 		if (user.stat != 0 || user.getStatusDuration("stunned"))
 			return
 		src.set_open(1)
+		sleep(20)
```
Comment (ZeWaka) [inline code/modules/vending/clothingbooth.dm:65]:
```suggestion
		sleep(2 SECONDS)
```

**Comment by flrsh** [comment]
Change all the typechecks for humans to just use the `ishuman(x)` macro. Also, probably output some sort of message for nonhumans when they try to interact with this. 

Also, I think generating flat icons is expensive, and that's why it was pulling from the saved `flat_icon` if it existed. I know the reason against pulling from `preview_icon` on preferences was that it might not pick your current character if a different slot were open, but I dunno if people ever actually even use the Character Setup command while ingame. Maybe it'd be a worthwhile tradeoff to just axe that, I'm not sure.

**Comment by AwkwardDryad** [comment]
> Change all the typechecks for humans to just use the `ishuman(x)` macro. Also, probably output some sort of message for nonhumans when they try to interact with this.
> 
> Also, I think generating flat icons is expensive, and that's why it was pulling from the saved `flat_icon` if it existed. I know the reason against pulling from `preview_icon` on preferences was that it might not pick your current character if a different slot were open, but I dunno if people ever actually even use the Character Setup command while ingame. Maybe it'd be a worthwhile tradeoff to just axe that, I'm not sure.

I removed the flat icon check, im using 100% preferences since this request was opened.
also ill definitely update ishuman :)

**Comment by ZeWaka** [comment]
If you want to fix travis failing, you'll have to pull from master.

File: code/modules/vending/clothingbooth.dm (hunk 1 above)
Comment (ZeWaka) [inline code/modules/vending/clothingbooth.dm:65]:
```suggestion
		sleep(2 SECONDS)
```

**Comment by Gerhazo** [comment]
:P as an emoji in direct boutputs to the player seems a tad too out of place/quirky. Perhaps just leave it at the information the user needs?

File: code/modules/vending/clothingbooth.dm [hunk 2]
```diff
@@ -146,7 +151,10 @@ var/list/clothingbooth_items = list()
 	src.open = new_open
 
 /obj/machinery/clothingbooth/attack_hand(mob/user as mob)
-	if(!user)
+	if(!ishuman(user))
+		boutput(user,"<span style=\"color:red\">Human clothes don't fit you, silly :P</span>")
+		return
+	if(!(user in range(1,src)))
```
Comment (ZeWaka) [inline code/modules/vending/clothingbooth.dm:157]: DIST_CHECK here instead

**Comment by AwkwardDryad** [comment]
Fixed :heart:

File: code/modules/vending/clothingbooth.dm [hunk 3]
```diff
@@ -100,14 +100,19 @@ var/list/clothingbooth_items = list()
...
 			if(istype(usr.equipped(),/obj/item/spacecash))
 				var/obj/item/dummycredits = usr.equipped()
 				src.money += dummycredits.amount
 				dummycredits.amount = 0
 				qdel(dummycredits)
+				return
 			else
 				uisetup(usr)
+				return
 		..()
```
Comment (pali6) [inline code/modules/vending/clothingbooth.dm:116]: Won't not calling parent cause problems with stuff? Like not being able to... paint the booth with a paint can for example, or not being able to deconstruct it?

File: code/modules/vending/clothingbooth.dm [hunk 4]
```diff
@@ -65,6 +62,9 @@ var/list/clothingbooth_items = list()
 		if (user.stat != 0 || user.getStatusDuration("stunned"))
 			return
 		src.set_open(1)
+		sleep(2 SECONDS)
```
Comment (pali6) [inline code/modules/vending/clothingbooth.dm:65]: I'm not sure if calling `sleep` in relaymove is a good idea.

File: code/modules/vending/clothingbooth.dm [hunk 5]
```diff
@@ -27,16 +27,13 @@ var/list/clothingbooth_items = list()
...
-	var/mob/living/carbon/human/H = user
-	if(istype(H) && H.flat_icon)
-		user << browse_rsc(H.flat_icon, "previewimage.png")
-	else
-		user.client.preferences.AH = user.bioHolder.mobAppearance
-		user.client.preferences.update_preview_icon()
-		user << browse_rsc(user.client.preferences.preview_icon, "previewimage.png")
+	user.client.preferences.AH = user.bioHolder.mobAppearance
+	user.client.preferences.update_preview_icon()
+	user << browse_rsc(user.client.preferences.preview_icon, "previewimage.png")
```
Comment (pali6) [inline code/modules/vending/clothingbooth.dm:36]: This will always cause it to use your first saved character instead of the one you are using currently.

Comment (AwkwardDryad) [reply code/modules/vending/clothingbooth.dm:36]: i was thinking about replacing it with a dummy player sprite mannequin for now actually.

File: code/modules/vending/clothingbooth.dm (hunk 4 above)
Comment (AwkwardDryad) [reply code/modules/vending/clothingbooth.dm:65]: ill look into it :heart:

File: code/modules/vending/clothingbooth.dm (hunk 3 above)
Comment (AwkwardDryad) [reply code/modules/vending/clothingbooth.dm:116]: i will also look into this. (sorry debugging my first patch is **AAAAAAAAAAAA**)

File: code/modules/vending/clothingbooth.dm (hunk 5 above)
Comment (pali6) [reply code/modules/vending/clothingbooth.dm:36]: That sounds like the best solution probably!

File: code/modules/vending/clothingbooth.dm (hunk 3 above)
Comment (AwkwardDryad) [reply code/modules/vending/clothingbooth.dm:116]: This should be fine as is. Though others can change it later if needed.

File: code/modules/vending/clothingbooth.dm (hunk 5 above)
Comment (AwkwardDryad) [reply code/modules/vending/clothingbooth.dm:36]: as it is, it's better than being your last built flat icon that's facing left and wearing a bunch of clothes :P

**Pull request merged**